Test/Optimizer/test_sr_onthefly.py

```diff
@@ -0,0 +1,192 @@
+import pytest
+from jax.config import config
```

Comment:
You have failing tests in the runner with no jax installed. 

Jax it's not an hard dependency on netket, and related functionality is only enabled when jax is detected. IIRC There is a bool flag in `netket.utils.jax_available` that you should use for that.

Likewise, tests should import jax only if this flag is true.

You can do something along the lines of [this](https://docs.pytest.org/en/latest/skipping.html), see the code below (untested). 

```
import netket as nk
import pytest


if not nk.utils.jax_available:
    pytest.skip("skipping jax-only SR-onthefly tests", allow_module_level=True)

import jax
etc...
```